packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -97,6 +110,55 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
     return null;
   }
 
+  const handleTestQuery = async () => {
+    setTestQueryResults(null);
+    setQueryError(null);
+
+    const value: ExposureQuery = {
+      name: exposureQuery.name,
+      query: userEnteredQuery,
+      id: dataSource.id,
```

Comment:
If this object is supposed to represent an exposure query, it shouldn't have the datasource id here.  I mentioned above, but since id isn't really needed to get the required columns, if we change that method signature to only take the required fields you can avoid this issue.